src/api/c/index.cpp

```diff
@@ -36,6 +36,38 @@ static void indexArray(af_array &dest, const af_array &src, const unsigned ndims
     dest = getHandle(dst);
 }
 
+af_err af_make_array_index(af_index_t** result, const af_array in)
+{
+    try {
+        if(*result) {
+            // avoid a dangling pointer
+            return AF_ERR_NOT_SUPPORTED;
+        }
+        *result = new af_index_t;
+        (*result)->idx.arr = in;
+        (*result)->isBatch = true;
```

Comment:
Hmm, I was under the impression that array_index has `isBatch = true;` & seq_index has `isSeq = true;' , i.e. this is mainly here to tell what is in the union struct.